src/NuGet.Core/NuGet.Packaging/PackageExtraction/PackageHelper.cs

```diff
@@ -37,16 +36,19 @@ public static bool IsPackageFile(string packageFileName, PackageSaveMode package
                 // This is to ignore archive entries that are not really files
                 return false;
             }
-            if (packageSaveMode.HasFlag(PackageSaveMode.Nuspec))
+
+            if (IsManifest(packageFileName))
```

Comment:
I would opt to go with the v3 approach that is a lot more predictable. I'm going to ping @MarkOsborneMS to verify this will not be a true blocker